### DISCUSSION TITLE
Fixed DataTable Header feature request

### DISCUSSION THREAD

**Discussion by DreadfulCode** [primefaces/community#2066] [PrimeVue]
Looking through https://primevue.org/datatable/ I don't see a setting for a sticky datatable header. 

If this feature does not exist yet I would like to request adding a new datatable parameter, such as "fixed". When set, the header of a datatable would stick to the top of the screen.

Any suggestions on some techniques to do this in lieu of this setting would be appreciated, too. 

Thanks!

**Comment by adholland** [comment]
Hi @DreadfulCode 

I had this same issue last week and solved it with the following.
1) Include the following css

.stickyTable.p-datatable > .p-datatable-wrapper {
  overflow: visible !important;
}

.stickyTable.p-datatable > .p-datatable-wrapper > .p-datatable-table > .p-datatable-thead {
  position: sticky;
  top: 0px;
  z-index: 10;
}

2. Add the stickyTable class to your DataTable
<DataTable
            class="widget-table stickyTable"
            
That should be all you need.
